src/linker/Linker/Driver.cs

```diff
@@ -240,6 +240,10 @@ public void Run (ILogger customLogger = null)
 								disabled_optimizations.Add (opt);
 
 							continue;
+
+						case "--ref":
+							context.Resolver.AddAssemblyPath (GetParam ());
```

Comment:
I don't understand why we need a different name for something what `-d <path>` already covers

Reply:
Something like `--ref` is necessary to prevent issues with`-d <path>` where different directories have dlls with the same name, but we only want to use one.

Specifically:
Our SDK passes the linker task a specific list of files as input. One of these is `WindowsBase.dll` from a netcoreapp directory. Some other inputs (for WPF apps) live in a windowsdesktop directory, which also contains a `WindowsBase.dll` (that we don't want to use). Before linking, the SDK already does conflict resolution to pick the right input. However, when the linker gets the directory of both inputs, it can resolve the wrong `WindowsBase.dll` when it scans whole directories.

Reply:
Right so the intention is to --ref behave like
`--reference /user/lib/file.dll`? It would be good if the names in the code actually reflected that

Reply:
Yup! Which names? Maybe SearchAssemblyPaths -> SearchFilePaths?

Reply:
I'd start with the documentation, it should be
`-reference <file>  .....`

similarly, with the resolver entry, replace `AddAssemblyPath ` with `AddReferenceAssembly` or something like that

Reply:
Done!